docs/framework/setting.md

```diff
@@ -25,7 +25,9 @@ The "Settings" system developed incrementally:
 - v4.2 – Added public Settings API. Included support for CRUD'ing data from multiple sources (`civicrm_domain` and `civicrm_setting`). Migrated more settings data from `civicrm_domain` to `civicrm_setting`.
 - v4.3 to v4.6 – Incrementally migrate more settings data.
 - v4.7 – Added `Civi::settings()` and refactored init system. Finished migrating data from `civicrm_domain` to `civicrm_setting`.
+- v5.7 - Added `Civi::contactSettings()` as a facade to allow for managing of Contact Settings rather than domain level settings.
 - v5.8 - Added Generic CiviCRM settings form.
+- v5.21 - Deprecated the usage of `CRM_Contact_BAO_Setting::setItem` in favour of the `Civi::settings` and `Civi::contactSettings` functionality and the Setting API to set settings
```

Comment:
No - this was discouraged long long before 5.21 - we've told people to use the api since about 4.2

Reply:
(saying deprecated here implies it was ever encouraged or supported for external use)

Reply:
would something like issued e-notices or something instead of deprecated make more sense?

Reply:
I suppose we could say something like 
CRM_Contact_BAO_Setting::setItem has not been supported from outside the core code base for some years but as of 5.21 it will issue notices in preparation for removal in a subsequent version

Reply:
(actually it has never been supported I would argue)

Reply:
OK, so it was nominally deprecated, and 5.21 will start to emit warnings. So suggestion:

```suggestion
- v5.21 - `CRM_Contact_BAO_Setting::setItem()` will now emit deprecation warnings. Use `Civi::settings()`, `Civi::contactSettings()`, or the Setting API instead.
```